bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandControllerDiscoveryService.java

```diff
@@ -0,0 +1,141 @@
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.touchwand.internal.TouchWandBindingConstants;
+import org.openhab.binding.touchwand.internal.TouchWandBridgeHandler;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TouchWandControllerDiscoveryService} Discovery service for Touchwand Controllers.
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = false, configurationPid = "discovery.touchwand")
+@NonNullByDefault
+public class TouchWandControllerDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int SEARCH_TIME_SEC = 2;
+    private static final int TOUCHWAND_BCAST_PORT = 35000;
+    private final Logger logger = LoggerFactory.getLogger(TouchWandControllerDiscoveryService.class);
+
+    private @Nullable Thread socketReceiveThread = null;
+    private DatagramSocket listenSocket;
+
+    public TouchWandControllerDiscoveryService() throws SocketException {
+        super(TouchWandBridgeHandler.SUPPORTED_THING_TYPES, SEARCH_TIME_SEC, true);
+        removeOlderResults(getTimestampOfLastScan());
+        listenSocket = new DatagramSocket(TOUCHWAND_BCAST_PORT);
+    }
+
+    @Override
+    protected void startScan() {
+        DatagramSocket localListenSocket = listenSocket;
+        runReceiveThread(localListenSocket);
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+    }
+
+    @Override
+    public void deactivate() {
+        if (socketReceiveThread != null) {
+            socketReceiveThread.interrupt();
+            socketReceiveThread = null;
+        }
+
+        listenSocket.close();
+        super.deactivate();
+    }
+
+    private void addDeviceDiscoveryResult(String label, String ip) {
+        String id = ip.replaceAll("\\.", "");
+        ThingUID thingUID = new ThingUID(THING_TYPE_BRIDGE, id);
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("label", label);
+        properties.put("ipAddress", ip);
+        // @formatter:off
+        logger.debug("Add new Bridge label:{} id {} ",label, id);
+        thingDiscovered(DiscoveryResultBuilder.create(thingUID)
+                .withThingType(THING_TYPE_BRIDGE)
+                .withLabel(label)
+                .withProperties(properties)
+                .withRepresentationProperty("ipAddress")
+                .build()
+        );
+        // @formatter:on
+    }
+
+    protected void runReceiveThread(DatagramSocket socket) {
+        Thread localSocketReceivedThread = socketReceiveThread = new ReceiverThread(socket);
+        localSocketReceivedThread.setName(TouchWandBindingConstants.DISCOVERY_THREAD_ID);
```

Comment:
There is a new thread naming scheme. See #8216 Can you set the name accordingly?